### PR TITLE
chore(parsers.csv): Add benchmark test

### PR DESCRIPTION
**Pull request by powersj**

### CODE REVIEW

**Comment by powersj** [comment]
@srebhan - I realized we had not talked about how to roll this out to other parsers. Was there a specific set of test data you wanted to use? Or if we have a test for the data + parse benchmark and a second for batch parsing if applicable is enough? Or should we be targeting other specific feature/options? Or add that as we need?

**Comment by Hipska** [comment]
@powersj also have a look at this RFC: #11850

File: plugins/parsers/csv/parser_test.go [hunk 1]
```diff
@@ -1515,3 +1515,65 @@ func TestParseCSVLinewiseResetModeAlways(t *testing.T) {
...
+			map[string]interface{}{
+				"value": 4,
+			},
+			time.Unix(0, 0),
+		),
+	}
+
+	actual, err := plugin.Parse([]byte(benchmarkData))
+	require.NoError(t, err)
+	testutil.RequireMetricsEqual(t, expected, actual, testutil.IgnoreTime())
```
Comment (Hipska) [inline plugins/parsers/csv/parser_test.go:1563]: You have the time in the csv, so the test should not ignore the timestamp..

**Review by srebhan — COMMENTED** [review]
@powersj I agree with @Hipska's comment on the timestamp. Other than that the code looks good.

**Review by Hipska — APPROVED** [review]

**Review by srebhan — APPROVED** [review]
Thanks @powersj!

**Pull request merged**